osu.Game/Utils/NamingUtils.cs

```diff
@@ -57,5 +58,19 @@ public static string GetNextBestName(IEnumerable<string> existingNames, string d
                 ? desiredName
                 : $"{desiredName} ({bestNumber})";
         }
+
+        /// <summary>
+        /// Given a set of <paramref name="existingFilenames"/> and a desired target <paramref name="desiredName"/>
+        /// finds a filename closest to <paramref name="desiredName"/> that is not in <paramref name="existingFilenames"/>
+        /// <remarks>
+        /// <paramref name="desiredName"/> SHOULD NOT CONTAIN the file extension.
+        /// </remarks>
+        /// </summary>
+        public static string GetNextBestFilename(IEnumerable<string> existingFilenames, string desiredName, string fileExtension)
+        {
+            var stripped = existingFilenames.Select(filename => filename.Substring(0, filename.Length - fileExtension.Length));
```

Comment:
The `.Substring()` call assumes that all extensions are of equal length and the same length as `fileExtension`. Which is generally not a good assumption to make in a generic helper method.

Please just use `Path.GetFileNameWithoutExtension()` as proposed above.